proposals/low-level-struct-improvements.md

```diff
@@ -50,248 +45,226 @@ readonly ref struct Span<T>
+It will be possible for such methods to exist. Specifically methods which take `ref` parameters, capture them in `ref` fields and return them. But they require a declarative opt-in to let the compiler, and developer, know that it is happening and to adjust the span spafety rules accordingly.
+
+It is important to understand these compat considerations before diving too far into the proposal here as they are central to parts of the design.
+
+## Detailed Design 
+The rules for `ref struct` safety are defined in the [span safety document](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.2/span-safety.md). This document will describe the required changes to this document as a result of this proposal. Once accepted as an approved feature these changes will be incorporated into that document.
+
+### Provide ref fields
+The language will allow developers to declare `ref` fields inside of a `ref struct`. This can be useful for example when encapsulating large mutable `struct` instances or defining high performance types like `Span<T>` in libraries besides the runtime.
+
+The set of changes to our span safety rules necessary to allow `ref` fields is small and targeted. However this section of the proposal is quite involved. The reason is to make sure the reader understands both the new rules and **why** they were chosen. There are a number of subtle interactions between `ref` fields and our [compat considerations](#compat-considerations) that are easy to miss. This section is meant to call these out and explain how they fit into the chosen rules.
 
-```cs
-class GoalOfRefFields
+The [new Span<T> definition](#new-span) also reveals several challenges that must be resolved for the lifetime of types that contain `ref` fields. They must both take into account the lifetime of captured values as well as the compat considerations.
+
+```c#
+Span<int> CreatingAndReturningSpan(ref int parameter)
 {
-    Span<T> CreateSpan<T>(ref T p)
-    {
-        // ERROR: the existing consumption rules for CreateSpan believe this 
-        // can never happen hence we must continue to enforce that it cannot
-        return new Span<T>(ref p);
+    int local = 42;
 
-        // Okay: this is legal today
-        return new Span<int>(new int[] { });
-    }
+    // This must be an error because the created Span<T> is storing a reference to the 
+    // stack. It cannot be safely returned.
+    return new Span<int>(ref local);
 
-    Span<int> ConsumptionCompatibility()
-    {
-        // Okay: this is legal today and must remain legal.
-        int local = 42;
-        return CreateSpan(ref local);
+    // This must be an error because of the compat issues of the existing span safety 
+    // rules.
+    return new Span<int>(ref parameter);
```

Comment:
i'm not really getting this statement (i feel like it hasn't been explained well enough).  Why is it necessary for something safe to be an error.  compat issues tell me that prior working code should continue to work.  but it doesn't tell my why something must be an error.

Reply:
I put a lot of the details into the earlier section but didn't have the full set of examples. I'm just going to expand the earlier section to include these, and link to that from here.